packages/docusaurus/src/server/utils.ts

```diff
@@ -31,3 +33,36 @@ export const logValidationBugReportHint = () => {
     )}\n`,
   );
 };
+
+// Recursively get the final routes (routes with no subroutes)
+export function getAllFinalRoutes(routeConfig: RouteConfig[]): RouteConfig[] {
+  function getFinalRoutes(route: RouteConfig): RouteConfig[] {
+    return route.routes ? flatMap(route.routes, getFinalRoutes) : [route];
+  }
+  return flatMap(routeConfig, getFinalRoutes);
+}
+
+export function reportMessage(
```

Comment:
I think this could be used for reporting things other than errors, so I named it in a general way.
I also prepended the type of message (info, warn, error) in front of the body so it looks similar to the other messages that we see in the cli (I've updated the screenshot in the original post)

Reply:
yes that looks fine